src/movies.js

```diff
@@ -1,22 +1,108 @@
+const movies = require("./data");
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(obj) {
+  let directores=[];
+  obj.map((obj)=>{
+      directores.push(obj.director)
+  })
+  return directores;
+}
+getAllDirectors(movies);
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(arr) {
+let contadorSteven=0;
+let numpelis= arr.filter((peli)=>
+  peli.director==='Steven Spielberg' && peli.genre.includes('Drama'));
+contadorSteven=numpelis.length
+return contadorSteven
+}
+
+howManyMovies(movies);
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(arr){
```

Comment:
Podemos retornar el acumulador si hpasan un array vacio ya que ese esta inicializado en 0
```suggestion
function scoresAverage(arr){
if(arr.length==0){
  return 0;
}
  let totalPoints=arr.reduce((acc,av)=>{
    let points= acc+=av.score
    if(arr.score){
      return points
     }else{
       return acc
     }

  },0);
  let average=(totalPoints/(arr.length)).toFixed(2)
 
  return Number(average)
    } ```